pkg/releaseutil/manifest_sorter.go

```diff
@@ -143,6 +143,10 @@ func (file *manifestFile) sort(result *result) error {
 
 		var entry SimpleHead
 		if err := yaml.Unmarshal([]byte(m), &entry); err != nil {
+			// Return a more understandable error message when the yaml fails to parse
+			if strings.Contains(err.Error(), "releaseutil.SimpleHead") {
```

Comment:
May you use something like `errors.As` or `errors.Is` instead of it?

Reply:
@wizbit just checking in on whether you had an answer for @cndoit18 's question.

Reply:
I don't know if it is possible. The error comes from the following stack

encoding/json/decode.go#938 `d.saveError(&UnmarshalTypeError{Value: "string", Type: v.Type(), Offset: int64(d.readIndex())})`
yaml.go#76 `return fmt.Errorf("while decoding JSON: %v", err)`
yaml.go#60 `return fmt.Errorf("error unmarshaling JSON: %v", err)`

My understanding of this is the `UnmarshalTypeError` is removed from the error in yaml.go such that only a string error exists.

Reply:
I checked this error chain and they don't support the wrapper.

It looks like the only way to compare is by strings.